sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2205,37 +2205,151 @@ class Dataset[T] private[sql](
    *   // max     92.0  192.0
    * }}}
    *
+   * See also [[summary]]
+   *
+   * @param cols Columns to compute statistics on.
+   *
    * @group action
    * @since 1.6.0
    */
   @scala.annotation.varargs
-  def describe(cols: String*): DataFrame = withPlan {
+  def describe(cols: String*): DataFrame = {
+    val selected = if (cols.isEmpty) this else select(cols.head, cols.tail: _*)
+    selected.summary("count", "mean", "stddev", "min", "max")
+  }
+
+  /**
+   * Computes specified statistics for numeric and string columns. Available statistics are:
```

Comment:
and explain the difference between describe and summary (basically summary seems easier to use).